packages/@react-native-windows/telemetry/src/utils/projectUtils.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ * @format
+ */
+
+import {createHash} from 'crypto';
+const fs = require('fs').promises; // Required to support Node 12
+import path from 'path';
+
+interface ProjectInfo {
+  id: string | null;
+  platforms: Array<string>;
+  rnwLang: 'cpp' | 'cs' | null;
+}
+
+export interface DependencyProjectInfo extends ProjectInfo {}
+
+export interface AppProjectInfo extends ProjectInfo {
+  usesTS: boolean;
+  usesRNConfig: boolean;
+  jsEngine: string;
+  rnwSource: string;
+  dependencies: Array<DependencyProjectInfo>;
+}
+
+/**
+ * Gets a unique, telemetry-safe project ID based on the project name.
+ * @param projectName The project name.
+ * @returns The telemetry-safe project ID.
+ */
+export function getProjectId(projectName: string): string {
+  const hash = createHash('sha256');
+  hash.update(projectName);
+  return hash.digest('hex');
+}
+
+/**
+ * Checks that a given file exits in the path specified.
+ * @param fileName The file to check for.
+ * @param projectRoot The root path to look in.
+ * @returns Whether the file exists.
+ */
+async function fileExists(
+  fileName: string,
+  projectRoot: string,
+): Promise<boolean> {
+  try {
+    const reactNativeConfigPath = path.resolve(projectRoot, fileName);
+    const stats = await fs.stat(reactNativeConfigPath);
+    return stats.isFile();
+  } catch {}
+  return false;
+}
+
+/**
+ * Gets whether the project at the given path has a react-native.config.js file.
+ * @param projectRoot The project root path to look in.
+ * @returns Whether the project at the given path has a react-native.config.js file.
+ */
+export async function usesReactNativeConfig(
+  projectRoot: string,
+): Promise<boolean> {
+  return fileExists('./react-native.config.js', projectRoot);
+}
+
+/**
+ * Gets whether the project at the given path is using TypeScript.
+ * @param projectRoot The project root path to look in.
+ * @returns Whether the project at the given path is using TypeScript.
+ */
+export async function usesTypeScript(projectRoot: string): Promise<boolean> {
+  return fileExists('./tsconfig.json', projectRoot);
+}
+
+/**
+ * Get the list of keys in the object for which the value is defined.
+ * @param obj The object to search.
+ * @returns The list of keys.
+ */
+function getDefinedKeys(obj: Record<string, any>): string[] {
+  return Object.keys(obj).filter(
+    value => obj[value] !== undefined && obj[value] !== null && value,
+  );
+}
+
+/**
+ * Given a react-native CLI config, determine the language of the RNW project if possible.
+ * @param config Config passed from react-native CLI.
+ * @returns The language of the RNW project.
+ */
+function getRnwLang(
+  config: Record<string, any> | null | undefined,
+): 'cpp' | 'cs' | null {
+  if (config) {
+    for (const project of config.projects) {
+      if (project.directDependency) {
+        return project.projectLang;
+      }
```

Comment:
this will return the first one, can't we have more than 1 direct dependency and so we should return an array of langs?

Reply:
Also, this just occurred to me, don't we want to include the non-direct dependencies' languages anyway? I mean, they're necessary for the project to build.

Reply:
I think that it's important that we capture the languages of every project that gets specified as needed. You can easily imagine a C# module project "wrapper" of a C++ non-module project, and we'd want to capture both.